### ISSUE TITLE
"restrict-plus-operands" doesn't understand BigInt

### ISSUE THREAD

**Issue by bowenni** [palantir/tslint#4741]
### Bug Report

- __TSLint version__: 5.16.0
- __TypeScript version__: 3.4.0
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
BigInt(1) + BigInt(2);
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "restrict-plus-operands": true
  }
}
```

#### Actual behavior
`Operands of '+' operation must either be both strings or both numbers, but found bigint + bigint`

#### Expected behavior
No errors.
